app/components/signup-steps/step-one.js

```diff
@@ -118,13 +113,28 @@ export default class SignupStepsStepOneComponent extends Component {
     }
   }
 
-  @action handleButtonClick() {
-    this.isSignupButtonDisabled = true;
-    this.signup();
-    localStorage.setItem('role', this.data.role);
-  }
-
   @action async signup() {
+    const { username } = await this.onboarding.generateUsername(
+      this.data.firstname,
+      this.data.lastname,
+    );
+
+    let dataToUpdate = {
+      username,
+      first_name: this.data.firstname,
+      last_name: this.data.lastname,
+    };
+
+    if (this.data.role !== 'Developer') {
+      dataToUpdate.roles = {
+        maven: this.data.role === 'Maven',
+        designer: this.data.role === 'Designer',
+        productmanager: this.data.role === 'Product Manager',
+      };
+    }
+
+    await this.onboarding.signup(dataToUpdate);
+    localStorage.setItem('role', this.data.role);
```

Comment:
This is being set here? Where is it being accessed?

Reply:
What is the purpose fo this

Reply:
Signup API is not taking a developer role so I have stored the role and based on role I have shown user different screen